src/base/helper.rs

```diff
@@ -29,3 +29,15 @@ where
     use std::iter;
     iter::repeat(()).map(|_| g.gen()).find(f).unwrap()
 }
+
+/// Check that the first and last offsets conform to the specification of a CSR matrix
+#[inline]
+#[must_use]
+pub fn first_and_last_offsets_are_ok(
+    major_offsets: &Vec<usize>,
+    minor_indices: &Vec<usize>,
+) -> bool {
+    let first_offset_ok = *major_offsets.first().unwrap() == 0;
+    let last_offset_ok = *major_offsets.last().unwrap() == minor_indices.len();
+    return first_offset_ok && last_offset_ok;
+}
```

Comment:
In general helper functionality should reside somewhere in `nalgebra-sparse`, since `nalgebra` itself does not have any knowledge of sparse matrices.